app/models/topic.rb

```diff
@@ -650,16 +649,13 @@ def add_moderator_post(user, text, opts=nil)
                               skip_validations: true,
                               custom_fields: opts[:custom_fields])
 
-    if (new_post = creator.create) && new_post.present?
-      increment!(:moderator_posts_count) if new_post.persisted?
-      # If we are moving posts, we want to insert the moderator post where the previous posts were
-      # in the stream, not at the end.
-      new_post.update_attributes!(post_number: opts[:post_number], sort_order: opts[:post_number]) if opts[:post_number].present?
+    new_post = creator.create
```

Comment:
Yes, that's the idea. After this set of changes, `PostCreator` when instantiated with the `skip_validations` parameter set to `true  should always allow instantiating an actual post and never return `nil`.